src/lib/autoinstall/autosetup_helpers.rb

```diff
@@ -197,8 +197,47 @@ def profile_checker
       Y2Autoinstallation::XmlChecks.instance
     end
 
+    # Invokes autoyast setup for firewall
+    def autosetup_firewall
+      return if !Yast::Profile.current["firewall"]
+
+      if need_second_stage_run?
+        # in some cases we need to postpone firewall configuration to the second stage
+        # we also have to guarantee that firewall is not blocking second stage in this case
+        fw_orig = Yast::Profile.current["firewall"]
+        Yast::Profile.current["firewall"] = { "enable_firewall" => false }
```

Comment:
Do not like this change, we are modifying the user profile, if you need to do it, would do in the the finish client..

Reply:
it is consequence of moving logic into autoyast from yast-firewall you asked for before.

So in the end is it so, that we need half of logic here and another half in the firewall module together with exporting this function?

I mean i need to do "something" in case when second stage is needed to guarantee that system remains reachable in the second stage. And if I have logic for handling optional first stage here even in yast2-firewall module I have to either duplicate the code or export this function so, which one is suitable for you.

Reply:
Ok, after discussing it, there is no way to handled it in the finish client without checking the current Profile which is something we are avoiding in general, so, the approach sounds good, maybe will just modify the code a little bit:

```ruby
# Invokes autoyast setup for firewall
def autosetup_firewall
  return if !Yast::Profile.current["firewall"]
    
  # in some cases we need to postpone firewall configuration to the second stage
  # we also have to guarantee that firewall is not blocking second stage in this case
  firewall_section =
    need_second_stage_run? ? { "enable_firewall" => false } : Yast::Profile.current["firewall"]
    
  
  log.info("Importing Firewall settings from AY profile")
  Yast::WFM.CallFunction("firewall_auto", ["Import", firewall_section])
  
  Yast::Profile.remove_sections("firewall") unless need_second_stage_run
end

```

Reply:
ok, will adapt and test again to check behavior

Reply:
adapted.